megatron/model/transformer.py

```diff
@@ -323,6 +375,21 @@ def __init__(self, init_method,
         self.attention_type = attention_type
         self.attn_mask_type = attn_mask_type
         self.params_dtype = args.params_dtype
+        self.sequence_parallel = args.sequence_parallel
+
+        self.use_flash_attn = args.use_flash_attn
+        if self.use_flash_attn:
+            if flash_attn_unpadded_func is None:
+                raise ImportError('FlashAttention is not installed, please install with '
+                                  'pip install flash-attn')
+            assert attention_type == AttnType.self_attn, ('FlashAttention code path only supports '
+                                                          'self-attention for now')
+            assert self.attn_mask_type == AttnMaskType.causal, ('FlashAttention code path only '
+                                                                'supports causal mask for now')
+            headdim = args.hidden_size / args.num_attention_heads
+            assert headdim <= 128, 'FlashAttention only supports head dimension at most 128'
```

Comment:
Does it make sense to put this assert in the flash-attn lib instead of here (or just rely on the one already there if it is there) so that if the lib changes in the future to support larger head dimensions we don't need to remember to update this code?

Reply:
Sounds good, I've removed the check here and we'll rely on the check in flash-attn itself.